doc/developer/design/20210408_timelines.md

```diff
@@ -0,0 +1,76 @@
+# Timelines
+
+## Summary
+
+Some sources consist of data that doesn't have any timestamp information and we
+assign that data "real time" timestamps. Other kinds of sources, might have stronger
+opinions about the timestamps that data get bound to. In those cases, source
+implementations have a tradeoff. If they remap upstream timestamps to "real time"
+then the data from this source can be joined against other real time sources easily.
+However, this makes doing other operations, like temporal filters, on the source
+more difficult because now timestamps within the data (like a "delete-at-this-time" column)
+might not be meaningful when compared to the timestamp Materialized assigned the data.
+
+This design doc proposes to:
+
+  * Introduce the concept of `source timeline`s or `source time domain` (name TBD).
+    A `timeline` means roughly the same thing it does colloquially when people say
+    for example "I wonder what it would be like in an alternate timeline where there
+    was no COVID". For our purpose more specifically, a `source timeline` means "a
```

Comment:
I fully acknowledge that this is a piss poor definition of timelines. Tried to think of a better one that didn't involve sets of sources, and could not. would appreciate suggestions

Reply:
I'm not sure I get this example, I didn't think of "timelines"/"time domains" being like, parallel universes, I think of them as "one process's beliefs about of the ordering of events," is that wrong? If we say that two sources are on the same timeline we're saying they have some kind of mutual understanding of how their events are ordered